woodwork/logical_types.py

```diff
@@ -638,3 +644,19 @@ def match(x):
             info = f"Series {series.name} contains invalid {type_string} values. "
             info += f"The {regex_key} can be changed in the config if needed."
             raise TypeValidationError(info)
+
+
+def _validate_not_negative(series, return_invalid_values):
+    """Validates data values are non-negative."""
+    invalid = series.lt(0)
+    if return_invalid_values:
+        return series[invalid]
+
+    else:
+        any_invalid = invalid.any()
+        if dd and isinstance(any_invalid, dd.core.Scalar):
+            any_invalid = any_invalid.compute()
+
+        if any_invalid:
+            info = f"Series {series.name} contains negative values."
+            raise TypeValidationError(info)
```

Comment:
If necessary, we can make the validation error message more specific to the logical type by making the message a parameter in the function